test/active_job/queue_adapters/adapter_testing/dispatch_jobs.rb

```diff
@@ -0,0 +1,15 @@
+module ActiveJob::QueueAdapters::AdapterTesting::DispatchJobs
+  extend ActiveSupport::Concern
+  extend ActiveSupport::Testing::Declarative
+
+  test "dispatch blocked job immediately" do
+    10.times { |index| BlockingJob.perform_later(pause: index * 0.1.seconds) }
+
+    blocked_jobs = ActiveJob.jobs.blocked
+    assert_equal 9, blocked_jobs.size
+
+    blocked_jobs.each(&:dispatch)
+    perform_enqueued_jobs
+    assert_empty blocked_jobs.reload
```

Comment:
I'd move this assertion to before `perform_enqueued_jobs`, because performing jobs will unblock the jobs. To verify this, check if the test passes if you remove this line: 
```ruby
 blocked_jobs.each(&:dispatch)
```
and if it does pass, it'd mean you're not quite testing the dispatching feature here.